imports/plugins/core/orders/server/no-meteor/queries/orders.js

```diff
@@ -0,0 +1,64 @@
+import escapeRegExp from "lodash/escapeRegExp";
+import ReactionError from "@reactioncommerce/reaction-error";
+
+/**
+ * @name orders
+ * @method
+ * @memberof Order/NoMeteorQueries
+ * @summary Query the Orders collection for orders and (optionally) shopIds
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} params - request parameters
+ * @param {String} params.accountId - Account ID to search orders for
+ * @param {String} params.orderStatus - Workflow status to limit search results
+ * @param {String} params.shopIds - Shop IDs for the shops that owns the orders
+ * @returns {Promise<Object>|undefined} - An Array of Order documents, if found
+ */
+export default async function orders(context, { filter, orderStatus, shopIds } = {}) {
+  const { collections, shopsUserHasPermissionFor, userHasPermission } = context;
+  const { Orders } = collections;
+
+  let query = {};
+
+  // If an admin wants all orders for an account, we force it to be limited to the
+  // shops for which they're allowed to see orders.
+  if (shopIds) {
+    shopIds.forEach((shopId) => {
+      if (!userHasPermission(["orders", "order/fulfillment"], shopId)) {
+        throw new ReactionError("access-denied", "Access Denied");
```

Comment:
It will stop the execution. This is how we do the other queries for orders and It makes sense. We shouldn't try to make the query work with bad data. If you put in 2 shops, and you don't have access to one, we should show an error so you don't have to guess that half of your dataset is missing.

In reality, we would limit the list of available shops for those users in the UI so the query would not fail from normal use.

Reply:
OK, makes sense